libcudacxx/test/libcudacxx/cuda/annotated_ptr/utils.h

```diff
@@ -64,6 +64,18 @@ __device__ __host__ __noinline__ T* alloc(bool shared = false)
   return arr;
 }
 
+template <typename T, int N>
+__device__ __noinline__ T* shared_alloc()
+{
+  __shared__ T data[N];
+
+  for (int i = 0; i < N; ++i)
+  {
+    data[i] = i;
+  }
+  return data;
+}
+
 template <typename T>
 __device__ __host__ __noinline__ void dealloc(T* arr, bool shared)
```

Comment:
Please also drop the `shared` code path from the other allocator

Reply:
Could be used elsewhere?

Reply:
I mean it is giving the wrong result so it should not be used elsewhere

Reply:
I made them distinct now.